### PR TITLE
Use pinned commit PyTorch on CI instead of nightly

### PR DESCRIPTION
**Pull request by huydhn**
I think this is a regression.  CI needs to use PyTorch built from the pinned commit instead of installing nightly (using `install_requirements.sh`).  Otherwise, both PT and ET CI won't be able to catch regression against their latest commits.

Various examples:

* https://github.com/pytorch/pytorch/pull/134319#issuecomment-2443209139 supposes to fail ET OSS test, but it passes because CI is running against PyTorch nightly, not the one built from the PR
* Issue encountered by  @tugsbayasgalan https://fb.workplace.com/groups/222849770514616/permalink/523920143740909 where CI runs against nightly instead of his PR

I attempt to fix the issue by instroducing `--use-pt-pinned-commit` to `install_requirements.sh`, so that it can keep the current behavior of installing pinned PT nightly when running locally while use pinned PT commit on CI.  Note that the Docker image only has the pinned PT commit installed.

### Testing

The correct PyTorch built from the pinned commit (23d590e518688f96e1d1947a08e9ca27df3e67e4) is used https://github.com/pytorch/executorch/actions/runs/11601294150/job/32303693643#step:12:6496

On the other hand, using nightly looks like https://github.com/pytorch/executorch/actions/runs/11600922135/job/32302497353#step:12:6552

### CODE REVIEW

**Review by kirklandsign — APPROVED** [review]

File: .ci/docker/requirements-ci.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 mpmath==1.3.0
-numpy==1.22.0; python_version == '3.10'
+numpy==1.21.3; python_version == '3.10'
```
Comment (guangy10) [inline .ci/docker/requirements-ci.txt:2]: Good catch. Thanks for making it match pyproject.toml https://github.com/pytorch/executorch/blob/main/pyproject.toml#L58-L60

File: install_requirements.py [hunk 2]
```diff
@@ -129,14 +145,25 @@ def python_is_compatible():
...
 
-# pip packages needed to run examples.
-# TODO: Make each example publish its own requirements.txt
-EXAMPLES_REQUIREMENTS = [
-    "timm==1.0.7",
-    f"torchaudio==2.5.0.{NIGHTLY_VERSION}",
-    "torchsr==1.0.4",
-    "transformers==4.42.4",
-]
+if USE_PYTORCH_NIGHTLY:
```
Comment (guangy10) [inline install_requirements.py:148]: Isn't it cleaner to have a single if block for the CI? And it should only apply to torch and the domain libraries, i.e. torch, torchvision, torchaudio?

**Review by guangy10 — APPROVED** [review]
Thanks for the CI improvement!

File: install_requirements.py [hunk 3]
```diff
@@ -111,12 +117,38 @@ def python_is_compatible():
...
-# pip packages needed by exir.
-EXIR_REQUIREMENTS = [
-    f"torch==2.6.0.{NIGHTLY_VERSION}",
-    f"torchvision==0.20.0.{NIGHTLY_VERSION}",  # For testing.
-    "typing-extensions",
-]
+if USE_PYTORCH_NIGHTLY:
+    # pip packages needed by exir.
+    EXIR_REQUIREMENTS = [
+        f"torch==2.6.0.{NIGHTLY_VERSION}",
```
Comment (guangy10) [inline install_requirements.py:123]:
```suggestion
        f"torch==2.6.0.{NIGHTLY_VERSION}" if USE_PYTORCH_NIGHTLY else "torch",
```
Maybe this is cleaner, so you don't have to specify other packages in both branches?

Comment (huydhn) [reply install_requirements.py:123]: Yeah, good idea, that looks nicer indeed

File: install_requirements.py [hunk 4]
```diff
@@ -113,11 +119,24 @@ def python_is_compatible():
 
 # pip packages needed by exir.
```
Comment (guangy10) [inline install_requirements.py:120]: Can you keep the comment for why the version is not needed when fetching nightly here?

Comment (huydhn) [reply install_requirements.py:120]: Oh I thought that I have mention it in the argument section above.  But I could add the comment back if you think it's closed to the code

**Review by guangy10 — APPROVED** [review]

**Pull request merged**